guvnor-structure/guvnor-structure-client/src/main/java/org/guvnor/structure/client/editors/repository/list/RepositoriesView.java

```diff
@@ -86,11 +87,14 @@ public boolean confirmDeleteRepository( final Repository repository ) {
 
     @Override
     public void removeIfExists( final Repository repository ) {
-        Widget w = repositoryToWidgetMap.remove( repository );
-        if ( w == null ) {
-            return;
+        Iterator<Map.Entry<Repository, Widget>> it = repositoryToWidgetMap.entrySet().iterator();
```

Comment:
@manstis After some debugging, that is the only possibility that I came up with too, but I haven't been able to see the motive why they are broken.